apps/spruce/src/pages/version/TestAnalysis/index.tsx

```diff
@@ -0,0 +1,254 @@
+                sendEvent({
+                  name: "Filtered test analysis tab",
+                  "filter.by": "test name",
+                });
+              }}
+              placeholder="Search failed tests (regex)"
+              validator={validateRegexp}
+            />
+            <Combobox
+              disabled={!hasResults}
+              label="Failure type"
+              multiselect
+              onChange={(value: string[]) => {
+                setSelectedTaskStatuses(value);
+                sendEvent({
+                  name: "Filtered test analysis tab",
+                  "filter.by": "task statuses",
+                });
+              }}
+              placeholder="Select a task status"
+              value={selectedTaskStatuses}
+            >
+              {taskStatuses.map((status) => (
+                <ComboboxOption
+                  key={status}
+                  data-cy={`task-status-${status}-option`}
+                  displayName={taskStatusToCopy[status]}
+                  value={status}
+                />
+              ))}
+            </Combobox>
+            <Combobox
+              disabled={!hasResults}
+              label="Build Variant"
+              multiselect
+              onChange={(value: string[]) => {
+                setSelectedBuildVariants(value);
+                sendEvent({
+                  name: "Filtered test analysis tab",
+                  "filter.by": "build variant",
+                });
+              }}
+              overflow="scroll-x"
+              placeholder="Select a build variant"
+              value={selectedBuildVariants}
+            >
+              {buildVariants.map((bv) => (
+                <ComboboxOption
+                  key={bv.buildVariant}
+                  data-cy={`build-variant-${bv.buildVariant}-option`}
+                  displayName={bv.buildVariantDisplayName}
+                  value={bv.buildVariant}
+                />
+              ))}
+            </Combobox>
+          </FilterContainer>
+          {hasResults && (
+            <FilterSubheaderContainer>
+              <Body weight="medium">
+                {totalFilteredTestCount}/{totalTestCount} Failed{" "}
+                {pluralize("Test", totalFilteredTestCount)}
+              </Body>
+              <Button
+                disabled={!hasFiltersApplied}
+                onClick={() => {
+                  setSelectedTaskStatuses([]);
+                  setSelectedBuildVariants([]);
+                  setTestName("");
+                }}
+                size="xsmall"
+              >
+                Clear Filters
+              </Button>
+            </FilterSubheaderContainer>
+          )}
+
+          <GroupedTestMapList groupedTestsMapEntries={groupedTestsMapEntries} />
+          {!hasMatchingResults && (
+            <BasicEmptyState
+              description=""
+              title={`No ${hasResults ? "Matching " : ""}Failed Tests Found`}
```

Comment:
[nit] Sentence case preferred
```suggestion
              title={`No ${hasResults ? "matching " : ""}failed tests found`}
```